tests/yam/agama/boot_agama.pm

```diff
@@ -16,6 +16,7 @@ use Utils::Backends;
 
 use Mojo::Util 'trim';
 use File::Basename;
+use Yam::Agama::agama_base 'upload_agama_logs';
```

Comment:
we could use Yam::Agama::agama_base as base class, but could you check perhaps how to avoid to save empty logs, something like check in upload_browser_automation_dumps that the folder doesn't exist for example.

Reply:
I did focus on just `upload_agama_logs` on purpose. If you look at the code of the module, all until `$agama_up_an_running->expect_is_shown();` is preparing the Grub setup. If we fail in this part, there is absolutely no kernel running on the SUT and therefore no way to retrieve any log. 

So if we fail to recognize that agama is up and running we at least have an environment to execute the log saving routine. But there won't be any browser automation dumps because we're still in the boot phase and no Puppeteer based code is executed on the SUT. 

So its easier to focus on just uploading what is there from agama instead of adding conditions to `agama_base` to find out if logs exist or not.

Reply:
ok, nevermind about empty logs, maybe that is a behavior of accepted in openQA (but a surprise for developers to find empty logs).
anyway we can try to upload all kind of logs, we missing journal logs, I don't know why not to add it as a base class, actually you don't know what is there, if the automation fails before reaching agama we also update something empty about agama logs I guess.

Reply:
`upload_agama_logs` stores a boot log with `journacltl -b` as well as it does `agama logs store` to provide `tar.gz` with the following contents:
```
$ tar tf boot_agama-agama-logs.tar.gz 
agama-logs.cx7uLg/
agama-logs.cx7uLg/rpm-qa.out.log
agama-logs.cx7uLg/dmesg.out.log
agama-logs.cx7uLg/agama-dbus-monitor.out.log
agama-logs.cx7uLg/agama-web-server.out.log
agama-logs.cx7uLg/agama-auto.out.log
agama-logs.cx7uLg/agama.out.log
agama-logs.cx7uLg/etc/
agama-logs.cx7uLg/etc/os-release
agama-logs.cx7uLg/run/
agama-logs.cx7uLg/run/agama/
agama-logs.cx7uLg/run/agama/dbus.log
agama-logs.cx7uLg/var/
agama-logs.cx7uLg/var/log/
agama-logs.cx7uLg/var/log/pbl.log
agama-logs.cx7uLg/var/log/YaST2/
agama-logs.cx7uLg/var/log/YaST2/y2log
agama-logs.cx7uLg/var/log/YaST2/config_diff_2024_12_16.log
agama-logs.cx7uLg/var/log/YaST2/storage-inst/
agama-logs.cx7uLg/var/log/YaST2/storage-inst/02-proposed.yml
agama-logs.cx7uLg/var/log/YaST2/storage-inst/01-probed.yml
agama-logs.cx7uLg/var/log/YaST2/storage-inst/02-proposed.xml
agama-logs.cx7uLg/var/log/YaST2/storage-inst/01-probed.xml

```
So again we have a  current `dmesg` ouit as well as the y2log and all the messages from the agama webserver if there are any. 

So what journal logs are still missing?